### PR TITLE
Fix error-prone warnings for io/hadoop-inputformat 

### PR DESCRIPTION
**Pull request by timrobertson100**
Really simple PR to fix error-prone and most static analysis warnings in io/hadoop-inputformat

CC @iemejia

### CODE REVIEW

**Comment by iemejia** [comment]
R: @jasonkuster by randomly assigning reviewers for the fixes on error-prone.
(and to make you use your new committer super powers).

File: sdks/java/io/hadoop-input-format/src/main/java/org/apache/beam/sdk/io/hadoop/inputformat/HadoopInputFormatIO.java [hunk 1]
```diff
@@ -234,6 +235,7 @@
     }
 
     /** Reads from the source using the options provided by the given configuration. */
+    @SuppressWarnings("unchecked")
```
Comment (jasonkuster) [inline sdks/java/io/hadoop-input-format/src/main/java/org/apache/beam/sdk/io/hadoop/inputformat/HadoopInputFormatIO.java]: Can we scope this (and others which are necessary) to just the line which is causing them, rather than putting them at the method level?

File: sdks/java/io/hadoop-input-format/src/main/java/org/apache/beam/sdk/io/hadoop/inputformat/HadoopInputFormatIO.java [hunk 2]
```diff
@@ -411,6 +414,7 @@ private void validateTranslationFunction(TypeDescriptor<?> inputType,
           null);
     }
 
+    @SuppressWarnings("WeakerAccess")
```
Comment (jasonkuster) [inline sdks/java/io/hadoop-input-format/src/main/java/org/apache/beam/sdk/io/hadoop/inputformat/HadoopInputFormatIO.java]: Why is this necessary?

**Comment by jasonkuster** [comment]
Thanks for the forward Ismael -- excited to be able to flex my committer muscles. :stuck_out_tongue:

File: sdks/java/io/hadoop-input-format/src/test/java/org/apache/beam/sdk/io/hadoop/inputformat/HIFITestOptions.java [hunk 3]
```diff
@@ -24,7 +24,7 @@
 /**
  * Properties needed when using HadoopInputFormatIO with the Beam SDK.
  */
-public interface HIFITestOptions extends TestPipelineOptions {
+interface HIFITestOptions extends TestPipelineOptions {
```
[the remark this answers is not in the document]
Comment (timrobertson100) [reply sdks/java/io/hadoop-input-format/src/test/java/org/apache/beam/sdk/io/hadoop/inputformat/HIFITestOptions.java]: In this case it is only used in the `DirectRunner` for the test so isn't referenced elsewhere.  Given it is within the tests, I opted to scope strictly but can revert this if you want?

File: sdks/java/io/hadoop-input-format/src/main/java/org/apache/beam/sdk/io/hadoop/inputformat/HadoopInputFormatIO.java (hunk 1 above)
Comment (timrobertson100) [reply sdks/java/io/hadoop-input-format/src/main/java/org/apache/beam/sdk/io/hadoop/inputformat/HadoopInputFormatIO.java]: I can do.  I'm cleaning many IO, and opted to put them on methods when there was >1 usages within the method.  In some IO I found 8 usages within one method which I felt annoyingly cluttered. There are two in this case.

Do you want them on a per line basis? Happy to change for this review.

File: sdks/java/io/hadoop-input-format/src/main/java/org/apache/beam/sdk/io/hadoop/inputformat/HadoopInputFormatIO.java (hunk 2 above)
Comment (timrobertson100) [reply sdks/java/io/hadoop-input-format/src/main/java/org/apache/beam/sdk/io/hadoop/inputformat/HadoopInputFormatIO.java]: IDEA code analysis will recommend this be `package-private`.  Because it is a `public static class` with no `final` and the author opted for `protected` I was cautious and opted not to change the scope as perhaps there is an idea it might be extended one day. I wanted to suppress the static analysis warning though.

Comment (timrobertson100) [reply sdks/java/io/hadoop-input-format/src/main/java/org/apache/beam/sdk/io/hadoop/inputformat/HadoopInputFormatIO.java]: IDEA code analysis will recommend this be `package-private`.  Because it is a `public static class` with no `final` and the author opted for `protected` for this method, I was cautious and opted not to change the scope as perhaps there is an idea it might be extended one day. I wanted to suppress the static analysis warning though.

**Comment by timrobertson100** [comment]
Thanks @tengpeng - I will do now I see we prefer Jiras for this.

**Comment by timrobertson100** [comment]
@jasonkuster - I am really sorry.
In a hasty move I stupidly rebased and pulled in 71 commits. I'm sorry about that.

Given this is a small one, I am going to close this and start afresh referencing the jira which now appears.
